src/shared/lib/hooks/useOnForegroundDebounced.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useRef } from 'react';
+import { AppState, AppStateStatus } from 'react-native';
+
+import { useDebouncedCallback } from 'use-debounce';
+
+type Options = {
+  enabled: boolean;
+};
+
+const DebounceInterval = 100;
+
+function useOnForegroundDebounced(
+  callback: () => void,
+  options?: Partial<Options>,
+) {
+  const callbackRef = useRef(callback);
```

Comment:
No need to use `useRef` for the callback here as `useDebouncedCallback` already does it under the hood